src/components/generics/date-range-picker.tsx

```diff
@@ -0,0 +1,98 @@
+import { Box } from "@mui/material";
+import { DatePicker } from "@mui/x-date-pickers";
+import { DateTime } from "luxon";
+import { useEffect, useState } from "react";
+import { getTimeDifferenceInDays } from "../../utils/time-utils";
+
+/**
+ * Component properties
+ */
+interface DateRangePickerProps {
+  dateTimeNow: DateTime;
+  setDates: Function;
+  initialStartDate?: DateTime;
+  initialEndDate?: DateTime;
+}
+/**
+ * Date range picker component
+ *
+ * @param props DateRangePickerProps
+ */
+const DateRangePicker = (props: DateRangePickerProps) => {
+  const { dateTimeNow, setDates, initialStartDate, initialEndDate } = props;
+  const [startDate, setStartDate] = useState<DateTime | null | undefined>(
+    initialStartDate ? initialStartDate : dateTimeNow
+  );
+  const [endDate, setEndDate] = useState<DateTime | null | undefined>(
+    initialEndDate ? initialEndDate : dateTimeNow
+  );
+
+  /**
+   * Set startDate and endDate on initial dates changing
+   */
+  useEffect(() => {
+    if (initialStartDate) {
+      setStartDate(initialStartDate);
+    } else {
+      setStartDate(dateTimeNow);
+    }
+    if (initialEndDate) {
+      setEndDate(initialEndDate);
+    } else {
+      setEndDate(dateTimeNow);
+    }
+  }, [initialStartDate, initialEndDate]);
+
+  /**
+   * Handle date change
+   *
+   * @returns startDate, endDate, days
+   */
+  const handleDateChange = (
+    startDate: DateTime | null | undefined,
+    endDate: DateTime | null | undefined
+  ) => {
+    let days;
+    let tempEndDate = endDate;
+    if (startDate && endDate) {
+      if (startDate > endDate) {
+        tempEndDate = startDate;
+      }
+      if (startDate === endDate) {
+        days = 1;
+      } else {
+        days = getTimeDifferenceInDays(startDate, endDate);
+      }
+    }
+    setStartDate(startDate);
+    setEndDate(tempEndDate);
+    setDates(startDate, tempEndDate, days);
+  };
```

Comment:
```suggestion
  const handleDateChange = (
    startDate: DateTime | null | undefined,
    endDate: DateTime | null | undefined
  ) => {
    if (startDate && endDate) {
      if (startDate > endDate) {
        setEndDate(startDate);
      }
      else setEndDate(endDate);
    }
    setStartDate(startDate);
    setDates(startDate, endDate, getTimeDifferenceInDays(startDate, endDate));
  };
```
Would this solution be simpler? I saw that getTimeDifferenceInDays returns 1 even if the end and start dates are same.

Reply:
Seems to be working the same. Committing this.